src/main/java/org/itsallcode/openfasttrace/importer/asciidoc/AsciiDocImporter.java

```diff
@@ -1,26 +1,267 @@
+                location));
+
+        this.listener.beginSpecificationItem();
+        this.listener.setId(specItemId);
+        this.listener.setLocation(location);
+        this.listener.addCoveredId(coveredSid);
+        processSpecificationItemNeeds(block);
+        this.listener.endSpecificationItem();
+    }
+
+    // [impl->dsn~adoc-specification-item-id~1]
+    // [impl->dsn~adoc-artifact-forwarding-notation~1]
+    private void processSpecificationItem(final StructuralNode block)
+    {
+        Optional.ofNullable(block.getAttribute(ATTRIBUTE_OFT_SID))
+                .filter(String.class::isInstance)
+                .map(String.class::cast)
+                .filter(s -> !s.isBlank())
+                .ifPresentOrElse(
+                        sid -> processSpecificationItemBlock(sid, block),
+                        () -> Optional.ofNullable(block.getAttribute(ATTRIBUTE_OFT_SKIPPED))
+                                .filter(String.class::isInstance)
+                                .map(String.class::cast)
+                                .filter(s -> !s.isBlank())
+                                .ifPresentOrElse(
+                                        skippedType -> processForwardingBlock(skippedType, block),
+                                        () -> LOG.severe(() -> """
+                                                Cannot process .specitem [%s]. A .specitem block must \
+                                                have either an 'oft-sid' or an 'oft-skipped' attribute.
+                                                """.formatted(getLocation(block)))));
     }
 
+    // [impl->dsn~adoc-specification-item-markup~1]
+    private Document parseAsciiDoc()
+    {
+        final Options options = Options.builder().sourcemap(true).build();
+        try (final Asciidoctor asciidoctor = Asciidoctor.Factory.create())
+        {
+            if (this.file != null)
+            {
+                LOG.fine(() -> "reading specification items from file: %s".formatted(this.file.toPath()));
+                return asciidoctor.loadFile(this.file.toPath().toFile(), options);
+            }
+            else if (this.content != null)
+            {
+                LOG.fine("reading specification items from document");
+                return asciidoctor.load(content, options);
+            }
+            else
+            {
+                // this cannot happen because the constructors require either
+                // file or document being provided
+                throw new IllegalStateException("either input file or document must be provided");
+            }
+        }
+    }
+
+    // [impl->dsn~adoc-specification-item-markup~1]
     @Override
     public void runImport()
     {
-        LOGGER.finest(() -> "Importing AsciiDoc file '" + file + "'");
+        final var document = parseAsciiDoc();
+
+        LOG.fine("processing specification item definitions");
```

Comment:
You might want to mention here that this is done by the AsciiDoc importer. Would help in the log stream.

Reply:
I have added this to the log statements in `parseAsciiDoc()`